freqtrade/rpc/rpc.py

```diff
@@ -1097,19 +1097,28 @@ def _rpc_delete(self, trade_id: int) -> dict[str, str | int]:
                 "cancel_order_count": c_count,
             }
 
-    def _rpc_list_custom_data(self, trade_id: int, key: str | None) -> list[dict[str, Any]]:
-        # Query for trade
-        trade = Trade.get_trades(trade_filter=[Trade.id == trade_id]).first()
-        if trade is None:
+    def _rpc_list_custom_data(self, trade_id: int | None = None, key: str | None = None) -> list[dict[str, Any]]:
+        # Query trades based on trade_id
+        if trade_id:
+            trades = Trade.get_trades(trade_filter=[Trade.id == trade_id]).all()
+        else:
+            # If no trade_id, get all trades
+            trades = Trade.get_trades().all()
```

Comment:
I'd not do this if i'm honest. 
this can yield responses for 1000nds of trades - creating an insanely large response.

if necessary - go with "all open trades" instead?

Reply:
Ok so ive updated this to use Trade.get_open_trades() ^^